### DISCUSSION TITLE
Some jquery migration questions

### DISCUSSION THREAD

**Discussion by shakazulu89** [serenity-is/Serenity#7099] [General]
Hey all

So wondering two questions:

1) What are the main reasons serenity is dropping jquery dependency?

2) Short background: I am migrating a large project from serenity 3.5!  What are the drawbacks of using "getJquery" and leaving all the jquery logic i have in typescript files like $.find, $.inarray, etc. etc.? 

Thanks for any feedback!

**Comment by VictorTomaili** [comment]
JQuery is a legacy library. It is used by many projects, but most modern applications don't prefer to use it because there are no additional benefits. And the modern browsers have a widely supported javascript api.
JQuery released an updated version after we decided to drop it. But this new version is not enough to continue to use.
With this, we eliminated a dependency from the project. This means less code to download from server to browser, less network traffic. Less global library. 

Using `getJquery` is designed for this reason. You can upgrade your project step by step and still continue to develop new features. But at the same time, you can prefer to update your existing code base. In a short time, I don't think we will drop the support of getJquery, because many of our customers still have Jquery dependency. And the transition needs more time.

**Comment by shakazulu89** [comment]
Right, figured as much just couldnt be sure of myself hehe .. thanks for the explanation Victor

One more question, probably a stupid one but ill ask anyways:

The example scenario is i have $ in many places in a page.
Is there a big performance difference in declaring one _$ = getjquery_ in class scope as opposed to replacing all $ with _getJquery()_? 

Obviously the latter will be easier to refactor but im just wondering whats the performance hit in your opinion?

**Reply by VictorTomaili** [reply]
getJquery is a function which contains additional conditions to check jquery is exists or not. Using it once will be a better option. But I don't think there are big performance effect. 
Totally depends on your decision.
